File: Java/lab11_Facade/task_2_2 - solution/src/Ignition.java
public class Ignition {

  private boolean isOn = false;

  public void turnOn() {
    System.out.println("Turning ignition on");
    isOn = true;
  }

  public void turnOff() {
    System.out.println("Turning ignition off");
    isOn = false;
  }

}

File: Cpp/lab11_Facade/task_2_2 - solution/src/ignition.cpp
#include <iostream>

#include "ignition.hpp"


Ignition::Ignition() {}

void Ignition::turnOn()
{
  std::cout << "Turning ignition on" << std::endl;
  this->isOn = true;
}

void Ignition::turnOff()
{
  std::cout << "Turning ignition off" << std::endl;
  this->isOn = false;
}
